PWGLF/TableProducer/LFTreeCreatorNuclei.cxx

```diff
@@ -0,0 +1,177 @@
+DECLARE_SOA_COLUMN(NSigTOF3He, nsigTOF3He, float);
+DECLARE_SOA_COLUMN(TOFmatch, tofMatch, bool);
+
+// Events
+DECLARE_SOA_COLUMN(IsEventReject, isEventReject, int);
+DECLARE_SOA_COLUMN(RunNumber, runNumber, int);
+} // namespace full
+
+DECLARE_SOA_TABLE(LfCandNucleusFull, "AOD", "LFNUCL",
+                  collision::BCId,
+                  full::NSigTPCPi,
+                  full::NSigTPCKa,
+                  full::NSigTPCPr,
+                  full::NSigTPCDe,
+                  full::NSigTPC3He,
+                  full::NSigTOFPi,
+                  full::NSigTOFKa,
+                  full::NSigTOFPr,
+                  full::NSigTOFDe,
+                  full::NSigTOF3He,
+                  full::TOFmatch,
+                  full::Px,
+                  full::Py,
+                  full::Pz,
+                  full::Pt,
+                  full::P,
+                  full::Eta,
+                  full::Phi,
+                  full::Sign);
+
+DECLARE_SOA_TABLE(LfCandNucleusFullEvents, "AOD", "LFNUCLEvent",
+                  collision::BCId,
+                  collision::NumContrib,
+                  collision::PosX,
+                  collision::PosY,
+                  collision::PosZ,
+                  full::IsEventReject,
+                  full::RunNumber);
+} // namespace o2::aod
+
+/// Writes the full information in an output TTree
+struct CandidateTreeWriter {
+  Produces<o2::aod::LfCandNucleusFull> rowCandidateFull;
+  Produces<o2::aod::LfCandNucleusFullEvents> rowCandidateFullEvents;
+
+  void init(o2::framework::InitContext&)
+  {
+  }
+
+  Configurable<float> yMin{"yMin", -0.5, "Maximum rapidity"};
+  Configurable<float> yMax{"yMax", 0.5, "Minimum rapidity"};
+  Configurable<float> cfgCutVertex{"cfgCutVertex", 10.0f, "Accepted z-vertex range"};
+  Configurable<float> cfgCutEta{"cfgCutEta", 0.8f, "Eta range for tracks"};
+  Configurable<float> nsigmacutLow{"nsigmacutLow", -8.0, "Value of the Nsigma cut"};
+  Configurable<float> nsigmacutHigh{"nsigmacutHigh", +8.0, "Value of the Nsigma cut"};
+
+  Filter collisionFilter = nabs(aod::collision::posZ) < cfgCutVertex;
+  Filter trackFilter = (nabs(aod::track::eta) < cfgCutEta) && (aod::track::isGlobalTrack == (uint8_t) true);
+  using TrackCandidates = soa::Filtered<soa::Join<aod::Tracks, aod::TracksExtra, aod::TracksExtended, aod::TrackSelection,
+                                                  aod::pidTPCFullPi, aod::pidTOFFullPi,
+                                                  aod::pidTPCFullKa, aod::pidTOFFullKa,
+                                                  aod::pidTPCFullPr, aod::pidTOFFullPr,
+                                                  aod::pidTPCFullDe, aod::pidTOFFullDe,
+                                                  aod::pidTPCFullHe, aod::pidTOFFullHe>>;
+
+  void process(soa::Filtered<soa::Join<aod::Collisions,
+                                       aod::EvSels>>::iterator const& collision,
+               TrackCandidates const& tracks)
```

Comment:
Here you could subscribe to BCs  `aod::BC const&` to have the info about the run number later

Reply:
See above.